credmark/cmf/types/contract.py

```diff
@@ -131,9 +134,51 @@ def _load(self):
             self._meta.constructor_args = res.get('constructor_args')
             self._meta.abi = ABI(res.get('abi'))
             self._meta.is_transparent_proxy = res.get('proxy', 0) == "1"
-            # TODO: Implementation needs to be validated on the db
+
             if self._meta.is_transparent_proxy:
-                self._meta.proxy_implementation = Contract(address=res.get('implementation'))
+                # TODO: as we only store the latest implementation in DB but not in the past.
+                proxy_address = res.get('implementation')
+                if context.chain_id == 1:
+                    # TODO: Implementation needs to be validated on the db
+                    if self._meta.contract_name in ['OwnedUpgradeabilityProxy']:
+                        if self.address == '0x0000000000085d4780B73119b644AE5ecd22b376':
```

Comment:
@leafyoung What are these magic addresses? (and 0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B below)

Reply:
They are all the variations of https://eips.ethereum.org/EIPS/eip-1967

I will refactor them into constants.

```
Logic contract address

Storage slot 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc (obtained as bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)).

Holds the address of the logic contract that this proxy delegates to.
```

Reply:
refactored into constants/function.